x-pack/plugins/alerting/server/alerts_client/lib/is_alert_improving.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get } from 'lodash';
+import type { Alert } from '@kbn/alerts-as-data-utils';
+import { ALERT_ACTION_GROUP } from '@kbn/rule-data-utils';
+import { Alert as LegacyAlert } from '../../alert';
+import { ActionGroup, AlertInstanceState, AlertInstanceContext, RuleAlertData } from '../../types';
+
+export const isAlertImproving = <
+  AlertData extends RuleAlertData,
+  LegacyState extends AlertInstanceState,
+  LegacyContext extends AlertInstanceContext,
+  ActionGroupIds extends string,
+  RecoveryActionGroupId extends string
+>(
+  alert: Alert & AlertData,
+  legacyAlert: LegacyAlert<LegacyState, LegacyContext, ActionGroupIds | RecoveryActionGroupId>,
+  actionGroups: Array<ActionGroup<string>>
+): boolean | null => {
+  const currentActionGroup = legacyAlert.getScheduledActionOptions()?.actionGroup;
+  const previousActionGroup = get(alert, ALERT_ACTION_GROUP);
+
+  if (!currentActionGroup || !previousActionGroup) {
+    return null;
+  }
+
+  // Get action group definitions
+  const currentActionGroupDef = actionGroups.find((ag) => ag.id === currentActionGroup);
+  const previousActionGroupDef = actionGroups.find((ag) => ag.id === previousActionGroup);
+  if (
+    currentActionGroupDef &&
+    previousActionGroupDef &&
+    currentActionGroupDef.severity &&
+    previousActionGroupDef.severity
+  ) {
+    const toRet =
+      currentActionGroupDef.severity.level === previousActionGroupDef.severity.level
```

Comment:
Makes sense! I can do that in this PR

Reply:
Added in https://github.com/elastic/kibana/pull/184163/commits/241e0063f7d71369b7fe3b365017b89f489c1e74

I added a check to verity no duplicate severity levels are defined but not a check to ensure every active action group has a severity level because the 2 rule types with multiple action groups both have an action group where severity doesn't make sense (`suppressed` for SLO rule type and `no_data` for metric threshold)